BridgeAppSDK/SBAProfileManager.swift

```diff
@@ -0,0 +1,239 @@
+public var SBAProfileItemsJSONFilename = "ProfileItems"
+public var SBAProfileManagerClassType = "ProfileManager"
+
+/**
+ Profile manager error types
+ */
+public enum SBAProfileManagerErrorType {
+    case unknownProfileKey
+}
+
+/**
+ Profile manager error object
+ */
+public class SBAProfileManagerError: NSObject, Error {
+    var errorType: SBAProfileManagerErrorType
+    var profileKey: String
+    
+    public init(errorType: SBAProfileManagerErrorType, profileKey key: String) {
+        self.errorType = errorType
+        self.profileKey = key
+        super.init()
+    }
+}
+
+public protocol SBAProfileManagerProtocol: NSObjectProtocol {
+
+    /**
+     Get a list of the profile keys defined for this app.
+     
+     @return A String array of profile item keys.
+     */
+    func profileKeys() -> [String]
+    
+    /**
+     Get the profile items defined for this app.
+     
+     @return A Dictionary of SBAProfileItem objects by profileKey.
+     */
+    func profileItems() -> [String: SBAProfileItem]
+    
+    /**
+     Get the value of a profile item by its profileKey.
+     
+     @return The value (optional) of the specified item.
+     */
+    func value(forProfileKey: String) -> Any?
+    
+    /**
+     Set the value of the profile item by its profileKey.
+     
+     @throws Throws an error if there is no profile item with the specified profileKey.
+     @param value The new value to set for the profile item.
+     @param key The profileKey of the item whose value is to be set.
+     */
+    func setValue(_ value: Any?, forProfileKey key: String) throws
+    
+    /**
+     Set up and return a view controller for displaying a Profile view.
+     
+     @return A view controller for displaying the Profile view.
+     */
+    func profileViewController() -> UIViewController?
+
+}
+
+
+open class SBAProfileManager: SBADataObject, SBAProfileManagerProtocol, SBAProfileDataSource {
+
+    /**
+     The shared instance of the profile manager. Loads from SBAProfileItemsJSONFilename, which
+     defaults to "ProfileItems".
+     */
+    static let shared: SBAProfileManagerProtocol? = {
+        guard let json = SBAResourceFinder.shared.json(forResource: SBAProfileItemsJSONFilename),
+                let sharedProfileManager = SBAClassTypeMap.shared.object(with:json, classType:SBAProfileManagerClassType) as? SBAProfileManagerProtocol
+            else { return nil }
+        return sharedProfileManager
+    }()
+
+    private dynamic var items: [SBAProfileItem] = []
+    lazy private var itemsKeys: [String] = {
+        var allKeys: [String] = []
+        for item in self.items {
+            allKeys.append(item.profileKey)
+        }
+        return allKeys
+    }()
+    
+    lazy private var itemsMap: [String: SBAProfileItem] = {
+        var allItems: [String: SBAProfileItem] = [:]
+        for item in self.items {
+            allItems[item.profileKey] = item
+        }
+        return allItems
```

Comment:
Just that mapping an array to a dictionary is something that I seem to do a lot.